File: frontend/src/types/article.ts
import { Summary } from "./summary";
import { Topic } from "./topic";

export interface Article {
  id: number;
  title: string;
  url: string;
  content: string;
  source: string;
  author: string;
  published_at: string;
  image_url: string;
  is_processed: boolean;
  topic_id: number;
  created_at: string;
  updated_at: string;
                
  topic?: Topic;
  summary?: Summary;

  processing_error?: string;
  source_metadata?: any;
}

export interface SearchParams {
  content?: string;
  title?: string;
  publishedAfter?: string;
  publishedBefore?: string;
  source?: string;
  topic: Topic | null;
  sortBy?: 'relevance'
}

File: backend/app/models/article.py
from sqlalchemy import Column, Integer, String, DateTime, Text, Boolean, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.core.database import Base

class Article(Base):
    __tablename__ = "articles"

    id = Column(Integer, primary_key=True, index=True)
    title = Column(String(500), nullable=False)
    url = Column(String(1000), nullable=False, unique=True)
    content = Column(Text, nullable=True) 
    source = Column(String(100), nullable=False)
    author = Column(String(200), nullable=True)
    published_at = Column(DateTime(timezone=True), nullable=True)
    image_url = Column(String(1000), nullable=True)
    is_processed = Column(Boolean, default=False)
    processing_error = Column(Text, nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())
    
    topic_id = Column(Integer, ForeignKey("topics.id", ondelete="CASCADE"), nullable=False)
    
    topic = relationship("Topic", back_populates="articles")
    summary = relationship("Summary", back_populates="article", uselist=False, cascade="all, delete-orphan")
